src/run.ts

```diff
@@ -234,13 +261,92 @@ async function destroy_stage(options: {
     });
   }
 
-  await destroyer.destroy(`${process.env.REGION_A}`, options.stage, {
+  const stacks = await getAllStacksForStage(
+    "us-east-1",
+    options.stage,
+    filters
+  );
+
+  const protectedStacks = stacks
+    .filter((i: any) => i.EnableTerminationProtection)
+    .map((i: any) => i.StackName);
+
+  if (protectedStacks.length > 0) {
+    console.log(
+      `We cannot proceed with the destroy because the following stacks have termination protection enabled:\n${protectedStacks.join(
+        "\n"
+      )}`
+    );
+    return;
+  } else {
+    console.log(
+      "No stacks have termination protection enabled. Proceeding with the destroy."
+    );
+  }
+
+  let notRetained: { templateKey: string; resourceKey: string }[] = [];
+  let retained;
+  if (
+    [
+      "main",
+      "master",
+      "val",
+      "production",
+      "jonsls1",
+      "jonsls2",
+      "jonsls3",
+      "jontester1",
+      "jontester2",
+      "jontester3",
+    ].includes(options.stage)
+  ) {
+    ({ retained, notRetained } = await checkRetainedResources(
+      options.stage,
+      filters
+    ));
+  }
+
+  if (retained) {
+    console.log("Information to use for import to CDK:");
+    retained.forEach(({ templateKey, resourceKey, physicalResourceId }) => {
+      // WaflogsUploadBucket is being retained but not being imported into CDK.
+      if (resourceKey !== "WaflogsUploadBucket") {
+        console.log(`${templateKey} - ${resourceKey} - ${physicalResourceId}`);
+      }
+    });
+  }
+
+  if (notRetained.length > 0) {
+    console.log(
+      "Will not destroy the stage because it's an important stage and some important resources are not yet set to be retained:"
+    );
+    notRetained.forEach(({ templateKey, resourceKey }) =>
+      console.log(` - ${templateKey}/${resourceKey}`)
+    );
+    return;
+  }
+
+  const accountId = await getAccountId();
+  await destroyer.destroy("us-east-1", options.stage, {
     wait: options.wait,
     filters: filters,
     verify: options.verify,
+    bucketsToSkip: [
+      `database-${options.stage}-mcpar`,
+      `database-${options.stage}-mlr`,
+      `database-${options.stage}-naaar`,
+      `ui-${options.stage}-cloudfront-logs-${accountId}`,
+      `${accountId}-ui-${options.stage}-waflogs`,
+    ],
   });
 
-  await delete_topics(options);
+  // await delete_topics(options);
```

Comment:
topics will be the same in cdk-land right?
This looks right just being sure I understand everything